src/server/rest/v1/service/BillingService.ts

```diff
@@ -644,8 +648,10 @@ export default class BillingService {
       Action.DOWNLOAD, Entity.BILLING, MODULE_NAME, 'handleDownloadInvoice');
     // Filter
     const filteredRequest = BillingSecurity.filterDownloadInvoiceRequest(req.query);
+    // Fetch tenant based on tenant ID
+    const tenant = await TenantStorage.getTenant(req.user.tenantID);
```

Comment:
Do not get the tenant, it's already provided in the HTTP request in 'req.tenant'.